packages/govuk-frontend/src/govuk/init.mjs

```diff
@@ -65,11 +79,30 @@ function initAll(config) {
  *
  * @template {CompatibleClass} T
  * @param {T} Component - class of the component to create
- * @param {T["defaults"]} [config] - config for the component
- * @param {Element|Document} [$scope] - scope of the document to search within
+ * @param {T["defaults"]} [config] - Config supplied to component
+ * @param {onErrorCallback<T> | Element | Document | createAllOptions<T> } [createAllOptions] - options for createAll including scope of the document to search within and callback function if error throw by component on init
  * @returns {Array<InstanceType<T>>} - array of instantiated components
  */
-function createAll(Component, config, $scope = document) {
+function createAll(Component, config, createAllOptions) {
+  let /** @type {Element | Document} */ $scope = document
+  let /** @type {onErrorCallback<Component> | undefined} */ onError
+
+  if (typeof createAllOptions === 'object') {
+    $scope =
+      /** @type {createAllOptions<Component>} */ (createAllOptions).scope ??
+      $scope
+    onError = /** @type {createAllOptions<Component>} */ (createAllOptions)
+      .onError
```

Comment:
**suggestion** Would it work to do something like the following to put the casting in its own line and make things a little more readable by separating the assignments of the options from the casting?

```suggestion
		createAllOptions = /** @type {createAllOptions<Component>} */ createAllOptions
		
    $scope = createAllOptions.scope ?? $scope
    onError = createAllOptions.onError
```

Reply:
That makes sense, I have disable eslint rule "no-self-assign" but it does look neater